x-pack/plugins/alerting/server/lib/flapping_utils.ts

```diff
@@ -5,31 +5,46 @@
  * 2.0.
  */
 
-const MAX_CAPACITY = 20;
-export const MAX_FLAP_COUNT = 4;
+import { RulesSettingsFlappingProperties } from '../../common/rules_settings';
 
-export function updateFlappingHistory(flappingHistory: boolean[], state: boolean) {
-  const updatedFlappingHistory = flappingHistory.concat(state).slice(MAX_CAPACITY * -1);
-  return updatedFlappingHistory;
+export function updateFlappingHistory(
+  flappingSettings: RulesSettingsFlappingProperties,
+  flappingHistory: boolean[],
+  state: boolean
+) {
+  if (flappingSettings.enabled) {
```

Comment:
We could set the status to always be false if flapping is disabled? Do you think that might be a better way to handle the flappingHistory array instead of clearing it out?

Reply:
`pendingRecoveredCount` does get reset in `get_alerts_for_notification.ts` if flapping is not enabled.

Reply:
Added flapping history to `get_alerts_for_notification.test.ts` in this commit 93f6043456a044857da13d958814c3cd4f755ffc

Reply:
> We could set the status to always be false if flapping is disabled? 

That sounds good.  But I'm still worried about "lingering" flapping history, during a disabled period.  I think that's kinda  a minor thing at this point, but still think we should open issue to track.  I'm guessing that rules that are flapping and then disabled, will still appear flapping, even if they aren't really, when re-enabled.  For a short while.

Reply:
Yes, we don't clear out flapping when a rule is disabled. I will open an issue!